hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientUserCodeDeploymentService.java

```diff
@@ -0,0 +1,170 @@
+        this.clientUserCodeDeploymentConfig = clientUserCodeDeploymentConfig;
+        this.configClassLoader = configClassLoader != null ? configClassLoader : ClassLoader.getSystemClassLoader();
+    }
+
+    public void start() throws IOException, ClassNotFoundException {
+        if (!clientUserCodeDeploymentConfig.isEnabled()) {
+            return;
+        }
+        loadClassesFromJars();
+        loadClasses();
+    }
+
+    private void loadClasses() throws ClassNotFoundException {
+        for (String className : clientUserCodeDeploymentConfig.getClassNames()) {
+            String resource = className.replace('.', '/').concat(".class");
+            InputStream is = null;
+            try {
+                is = configClassLoader.getResourceAsStream(resource);
+                if (is == null) {
+                    throw new ClassNotFoundException(resource);
+                }
+                byte[] bytes = toByteArray(is);
+                classDefinitionList.add(new AbstractMap.SimpleEntry<String, byte[]>(className, bytes));
+            } catch (IOException e) {
+                EmptyStatement.ignore(e);
+            } finally {
+                IOUtil.closeResource(is);
+            }
+        }
+    }
+
+    private void loadClassesFromJars() throws IOException {
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
+        try {
+            for (String jarPath : clientUserCodeDeploymentConfig.getJarPaths()) {
+                loadClassesFromJar(os, jarPath);
+            }
+        } finally {
+            IOUtil.closeResource(os);
+        }
+    }
+
+    private void loadClassesFromJar(ByteArrayOutputStream os, String jarPath) throws IOException {
+        JarInputStream inputStream = null;
+        try {
+            inputStream = getJarInputStream(jarPath);
+            JarEntry entry;
+            do {
+                entry = inputStream.getNextJarEntry();
+                if (entry == null) {
+                    break;
+                }
+
+                String className = extractClassName(entry);
+                if (className == null) {
+                    continue;
+                }
+                byte[] classDefinition = readClassDefinition(inputStream, os);
+                inputStream.closeEntry();
+                classDefinitionList.add(new AbstractMap.SimpleEntry<String, byte[]>(className, classDefinition));
+            } while (true);
+        } finally {
+            IOUtil.closeResource(inputStream);
+        }
+    }
+
+    private JarInputStream getJarInputStream(String jarPath) throws IOException {
+        File file = new File(jarPath);
+        if (file.exists()) {
+            return new JarInputStream(new FileInputStream(jarPath));
+        }
+
+        InputStream inputStream = configClassLoader.getResourceAsStream(jarPath);
+        if (inputStream == null) {
+            throw new FileNotFoundException("File could not be found in " + jarPath
```

Comment:
👍  I am adding URL handling to handle following usage
`addJar(url.toExternalForm())`